src/app/workflow/workflow-file-editor/workflow-file-editor.component.ts

```diff
@@ -77,23 +77,29 @@ export class WorkflowFileEditorComponent extends FileEditing {
    * Creates a new version based on changes made
    */
   saveVersion() {
-    const message = 'Save Version';
     const combinedSourceFiles = this.getCombinedSourceFiles();
     const newSourceFiles = this.commonSaveVersion(this.originalSourceFiles, combinedSourceFiles);
     this.alertService.start('Updating hosted workflow');
     this.hostedService.editHostedWorkflow(
-        this.id,
-        newSourceFiles).subscribe((workflow: Workflow) => {
-          this.toggleEdit();
+      this.id,
+      newSourceFiles).subscribe((workflow: Workflow) => {
+        this.toggleEdit();
+        if (workflow) {
           this.workflowsService.getWorkflow(workflow.id).subscribe((workflow2: Workflow) => {
             this.alertService.detailedSuccess();
             this.workflowService.setWorkflow(workflow2);
-          });
-        }, error =>  {
-          if (error) {
+          }, error => {
             this.alertService.detailedError(error);
-          }
+          });
+        } else {
+          // Probably encountered a 204
+          this.alertService.detailedSuccess('Version did not change');
```

Comment:
Please say why it didn't change (e.g. no file changes provided, new files are not different, etc.)